### ISSUE TITLE
Infinite 0s weakness caused by creeper explosion

### ISSUE THREAD

**Issue by Rad586** [sf-inc/sihywtcamd#44]
![image](https://github.com/sf-inc/sihywtcamd/assets/99306685/76fc916c-20f8-4a1e-accf-e5303ad096ef)
This bug happens occasionally when taking explosion damage from creepers in my modpack. 
Maybe a bug with health blast? (https://www.curseforge.com/minecraft/mc-mods/health-blast)
Mod list: https://gist.github.com/Rad586/696a472af82a0227c177b240743f99cb

**Comment by charlyb01** [comment]
Hello! Thanks for the report, I will try to reproduce and fix this.  
Does it happen only under certain circumstances or does it happen everytime?

**Comment by Rad586** [comment]
I reproduced it with only sihywtcamd: 

https://github.com/sf-inc/sihywtcamd/assets/99306685/172f0bf9-5f14-4393-98e0-68a48a6352f9

Nothing was logged in log.

**Comment by charlyb01** [comment]
Thanks a lot for the video, I really appreciate it! I will check and get back to you 😉

**Comment by charlyb01** [comment]
Hi again,

I thought that I found a way to reproduce each time, but now I struggle reproducing. However, I'm pretty sure I found what caused the issue. If you know how to reproduce easily and have time, I'm sharing with you a version in which I included the small fix. If you could take a look or at least trying to explain me what criteria you think causes the issue to reproduce.

Here is a 1.20.2 version of the mod with the fix:  
[sihywtcamd-1.8.2+1.20.2.zip](https://github.com/sf-inc/sihywtcamd/files/13551134/sihywtcamd-1.8.2%2B1.20.2.zip)

**Comment by Rad586** [comment]
It was fixed, I can't reproduce it now. 

I found this bug few months ago, but I didn't take it seriously, I thought it was all random, until I somehow reproduced it when running away from a creeper. The overlay only disappears when it's overlapped by another one, and it's flashing. So I think it has a low duration. 

I guess the problem comes from `Math.sqrt(1.0 - (this.distanceTo(entity) / explosionRadius));`. when distance＜radius, it results in nagetive value, and sqrt fails. There might be a tiny time difference to get the distance, causing it higher than radius. 
And maybe `final int explosionRadius = this.explosionRadius * (this.shouldRenderOverlay() ? 3 : 2);` is one reason for that, idk, maybe there's something going on with "shouldRenderOverlay". 

Not sure if I said it right, just for fun.
